extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java

```diff
@@ -155,6 +157,16 @@ public class WorkerImpl implements Worker
    */
   private volatile boolean controllerAlive = true;
 
+  /**
+   * Stores list of datasources queried for metric dimensions.
+   */
+  private final Set<String> datasources = new HashSet<>();
```

Comment:
Let's just have `datasources` and `intervals` combined into `Map<String, Object> queryMetricDimensions`. It's simpler and it avoids having "tempting" fields lying around like `datasources` and `intervals`. We don't want these used for any purpose other than metric emission.